assets/infrastructure-providers/infrastructure-azure.yaml

```diff
@@ -27,6 +27,7 @@ data:
           creationTimestamp: null
           labels:
             aadpodidbinding: capz-controller-aadpodidentity-selector
+            azure.workload.identity/use: "true"
```

Comment:
This label is used by the Azure pod identity webhook to source configurations set in ServiceAccount labels to env vars set for the pod. We didn't intend for the webhook to be used with core components since we have the details in the configuration secret generated by CCO. We may need to not set this somehow if it comes from `make assets`.

Reply:
May be okay to keep if the ServiceAccount doesn't ever receive the labels.

Reply:
This comes from `make assets` as well, afaik, a rework for the generation of the assets is being worked on. Right now, I am not sure whether we want to put it in there as it would break the `ci/prow/assets` job. WDYT, @JoelSpeed?
Same comes for the aforementioned secret variable path.

Reply:
Not sure why this new annotation would suddenly come in, was this added just by running `make assets` or was it manually added?

Reply:
I had to rerun assets generation, because it was broken. All the stuff was added by it.

Reply:
> I had to rerun assets generation, because it was broken.

What part was broken? Running the operator on Azure?

Reply:
Just the assets job was failing and had to rerun `make assets`.